src/main/java/book/store/model/User.java

```diff
@@ -19,7 +22,10 @@
 import org.springframework.security.core.userdetails.UserDetails;
 
 @Entity(name = "users")
-@Data
+@Getter
+@Setter
+@EqualsAndHashCode
```

Comment:
use this to avoid bugs related to auto-generating equals and hashcode for JPA Entities
```suggestion
@EqualsAndHashCode.of("id")
```